db/migrate/20171123163635_add_index_on_stream_entries.rb

```diff
@@ -0,0 +1,6 @@
+class AddIndexOnStreamEntries < ActiveRecord::Migration[5.1]
+  def change
+    commit_db_transaction
+    add_index :stream_entries, [:id, :account_id, :activity_type], algorithm: :concurrently
```

Comment:
I think you should put `id` at the last for sorting. By putting `account_id` at the very beginning you can also cover `index_stream_entries_on_account_id` and remove it.